src/transform.ts

```diff
@@ -575,31 +576,89 @@ export const transform = {
+        specifiers,
+        source,
+        leadingComments,
+        trailingComments,
+        loc,
+      } = path.node as ImportDeclaration;
+
       if (
-        path.node.importKind === "typeof" &&
-        t.isImportDefaultSpecifier(path.node.specifiers[0])
+        importKind === "typeof" &&
+        t.isImportDefaultSpecifier(specifiers[0])
       ) {
-        const {
-          specifiers,
-          source,
-          leadingComments,
-          trailingComments,
-          loc,
-        } = path.node;
-        const replacementNode = t.tsTypeAliasDeclaration(
+        replacementNode = t.tsTypeAliasDeclaration(
           specifiers[0].local,
           undefined,
           t.tsTypeQuery(t.tsImportType(source, t.identifier("default")))
         );
         replacementNode.leadingComments = leadingComments;
         replacementNode.trailingComments = trailingComments;
         replacementNode.loc = loc;
+      } else if (importKind === "value") {
+        // find any "type" imports within the ImportSpecifier list
+        // and pull them out to separate ImportDeclarations
 
-        trackComments(replacementNode, state);
+        const [valSpecs, typeSpecs] = partition(
+          specifiers,
+          (s): s is ImportSpecifier =>
+            s.type === "ImportSpecifier" && s.importKind === "type"
+        );
 
-        path.replaceWith(replacementNode);
-      } else {
-        trackComments(path.node, state);
+        if (typeSpecs.length > 0) {
+          const typeNode = t.importDeclaration(
+            typeSpecs.map((s) =>
+              t.importSpecifier(
+                returning(t.identifier(s.local.name), (n) => {
+                  n.loc = s.local.loc;
+                }),
+                returning(
+                  t.identifier(
+                    s.imported.type === "Identifier"
+                      ? s.imported.name
+                      : s.imported.value
+                  ),
+                  (n) => {
+                    n.loc = s.imported.loc;
+                  }
+                )
+              )
+            ),
+            source
+          );
+          typeNode.leadingComments = leadingComments;
+          typeNode.importKind = "type";
+          typeNode.loc = loc;
+
+          const valNode = t.importDeclaration(valSpecs, source);
+          valNode.trailingComments = trailingComments;
+          valNode.loc = {
+            start: {
+              line: loc.start.line + 1,
+              column: 0,
+            },
+            end: {
+              line: loc.start.line + 1,
+              column: 0,
+            },
+          };
+
+          replacementNode = [typeNode, valNode];
+        }
+      }
+
+      trackComments(replacementNode ?? path.node, state);
+
+      if (replacementNode != null) {
+        if (Array.isArray(replacementNode)) {
+          path.replaceWithMultiple(replacementNode);
+        } else {
+          path.replaceWith(replacementNode);
+        }
```

Comment:
style nit: It feels like drying up the code made things more complicated.  Could you move the `trackComments()` and `replaceWith()`/`replaceWithMultiple()` cals back into the if/else-if/else blocks?